### DISCUSSION TITLE
Output from Matrix / Supertable

### DISCUSSION THREAD

**Discussion by tigerchick** [trendyminds/craft-reporter#13] [Q&A] [answered]
Thank you for making this extremely useful plugin available.
Quick question: how should I best output from a matrix field (and / or a Supertable field - potentially within a Matrix field)?
Many thanks

**Comment by aaronbushnell** [comment, marked as answer]
Thanks so much, @tigerchick, that means a lot! So, ultimately, XLSX and CSVs do a pretty poor job parsing multi-dimensional content. That's why you're better off getting it into some kind of comma-separated list. Here's an example of an endpoint that outputs an example set of categories or Matrix blocks into Reporter:

```php
'transformer' => function (Entry $entry) {
    return [
        "id" => $entry->id,
        "title" => $entry->title,
        "url" => $entry->url,
        "categories" => $entry->categories->collect()
            ->pluck('title') // Category One, Category Two, ...
            ->join(', '),
        "socialNetworks" => $entry->matrixField->collect()
            ->map(fn ($block) => "$block->networkName ($block->url)") // Facebook (https://facebook.com/...), Twitter (https://twitter.com/...)
            ->join(', '),
    ];
}
```
